File: 0047-permutations-ii/0047-permutations-ii.cpp
class Solution {
public:
void permutaion(vector<int>& nums,vector<vector<int>>&ans,int idx){
    if(idx==nums.size()){
        ans.push_back(nums);
        return;
    }
    vector<bool>use(21,0);
    for(int i=idx;i<nums.size();i++){
        if(use[nums[i]+10]==0){
            swap(nums[i],nums[idx]);
            
            permutaion(nums,ans,idx+1);
            swap(nums[i],nums[idx]);
            use[nums[i]+10]=1;
        }
    }

}
    vector<vector<int>> permuteUnique(vector<int>& nums) {
        vector<vector<int>>ans;
        permutaion(nums,ans,0);
        return ans;
    }
};

File: 0047-permutations-ii/0047-permutations-ii.py
class Solution:
    def permuteUnique(self, nums: List[int]) -> List[List[int]]:
        res=[]
        def sol(idx,nums):
            if idx==len(nums):
                res.append(nums[:])
                return
            s=set()
            for i in range(idx,len(nums)):
                if nums[i] in s:
                    continue
                s.add(nums[i])
                nums[idx],nums[i]=nums[i],nums[idx]
                sol(idx+1,nums)
                nums[idx],nums[i]=nums[i],nums[idx]

        sol(0,nums)
        return res